tests/driver/driver_test.py

```diff
@@ -578,6 +598,23 @@ def test_finite_schedule_with_progress_indication(self):
             (4.0, metrics.SampleType.Normal, 5 / 5, {"body": ["a"], "size": 5}),
         ], list(invocations), infinite_schedule=False)
 
+    def test_schedule_with_progress_determined_by_runner(self):
+        task = track.Task("time-based", track.Operation("time-based", "driver-test-runner-with-completion",
+                                                        params={"body": ["a"]},
+                                                        param_source="driver-test-param-source"),
+                          clients=1,
+                          params={"target-throughput": 1, "clients": 1})
+
+        invocations = driver.schedule_for(self.test_track, task, 0)
+
+        self.assert_schedule([
+            (0.0, metrics.SampleType.Normal, None, {"body": ["a"]}),
+            (1.0, metrics.SampleType.Normal, None, {"body": ["a"]}),
+            (2.0, metrics.SampleType.Normal, None, {"body": ["a"]}),
+            (3.0, metrics.SampleType.Normal, None, {"body": ["a"]}),
+            (4.0, metrics.SampleType.Normal, None, {"body": ["a"]}),
+        ], invocations, infinite_schedule=True)
```

Comment:
I think that when we merge this to current master it's going to fail due to https://github.com/elastic/rally/pull/801/files

Reply:
Good catch! I've merged master and changed the test now in https://github.com/elastic/rally/pull/800/commits/78235be4769abbe88f0f456efcd7dcf81404defb.